src/cli/main.py

```diff
@@ -197,6 +199,60 @@ def get_job(ctx: click.Context, name: str) -> None:
     _output_result(resp)
 
 
+def make_job_create_request(
+    marketplace_image_urn: str,
+    vhd_sas_url: str,
+    job_name: str,
+    template_name: str,
+    resource_group: str,
+    subscription_id: str,
+    vm_size: str,
+    test_priorities: list[str],
+    test_cases: list[str],
+    location: str,
+    region: list[str],
+    concurrency: int,
+    vm_generation: int,
+    architecture: str,
+) -> tuple[dict[str, Any], str]:
+    endpoint = _get_endpoint(f"jobs/{job_name}", resource_group, subscription_id)
+
+    image = {
+        "vhdGeneration": vm_generation,
+        "architecture": architecture,
+    }
+    if marketplace_image_urn:
+        image["type"] = "marketplace"
+        image_parts = marketplace_image_urn.split(":")
+        if len(image_parts) != 4:
+            raise ValueError(
+                "Marketplace_image_urn should be in the format of 'publisher:offer:sku:version', "
+                f"got {marketplace_image_urn}"
+            )
+        image["publisher"] = image_parts[0]
+        image["offer"] = image_parts[1]
+        image["sku"] = image_parts[2]
+        image["version"] = image_parts[3]
+    elif vhd_sas_url:
+        image["type"] = "vhd"
+        image["url"] = vhd_sas_url
+    else:
+        raise ValueError("One of --vhd-sas-url or --marketplace-image-urn should be passed.")
```

Comment:
Should we raise an error when both are provided, or is silently allowing the Marketplace image URN to take precedence an acceptable user experience?

Reply:
If we want to default to something, I would default to the VHD URL. However, I don't like these kind of default behaviors. In this case, we don't know what the use wanted to do, so it's better to just print an error.

Reply:
Right, so this conditional logic should be restructured as:

```
if marketplace_image_urn and vhd_sas_url:
    raise ValueError(...)
elif marketplace_image_urn:
    ...
elif vhd_sas_url:
    ...
else:
   raise ValueError(...)
```

Reply:
Done